File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/SWModelList.java
package dev.radley.omgstarwars.models;

import java.io.Serializable;
import java.util.ArrayList;

/**
 * Generic list model
 */
public class SWModelList<T> implements Serializable {
    public int count;
    public String next;
    public String previous;
    public ArrayList<T> results;
}

File: OMGStarWars/app/src/main/java/dev/radley/omgstarwars/models/SWModelList.kt
package dev.radley.omgstarwars.models

import java.io.Serializable
import java.util.ArrayList

/**
 * Generic list model
 */
class SWModelList<T> : Serializable {
    var count: Int = 0
    var next: String? = null
    var previous: String? = null
    var results: ArrayList<T>? = null
}